### PR TITLE
Feature/delete areas

### PR DESCRIPTION
**Pull request by edward3sc**
New DELETE endpoint at /v3/forest-watcher/area/{areaId}.
Checks whether the logged-in user created the area. If not, checks whether the logged-in user is a manager of a team associated with the area. Deletes the record and associated Team and Template relations and returns a 204 if successful. Returns a 401 if unauthorised.

### CODE REVIEW

File: app/src/services/areas.service.js [hunk 1]
```diff
@@ -45,13 +45,34 @@ class AreasService {
...
     } catch (e) {
       logger.error("Error while fetching area", e);
       throw e;
     }
   }
 
+  static async delete(areaId) {
+    logger.info("Deleting area with id ", areaId);
+    try {
+      let baseURL = "https://api.resourcewatch.org/v2";
```
Comment (owenjs) [inline app/src/services/areas.service.js]: can this go in an environment variable?

File: app/src/services/team.service.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ class TeamService {
       const baseURL = config.get("teamsAPI.url");
```
Comment (owenjs) [inline app/src/services/team.service.js:10]: does this contain the new v3 team endpoint URL?

**Review by owenjs — CHANGES_REQUESTED** [review]

File: app/src/services/team.service.js (hunk 2 above)
Comment (edward3sc) [reply app/src/services/team.service.js:10]: Yes

File: app/src/services/areas.service.js (hunk 1 above)
Comment (edward3sc) [reply app/src/services/areas.service.js]: Done

File: terraform/vars/terraform-dev.tfvars [hunk 3]
```diff
@@ -12,6 +12,7 @@ areas_api_url             = "https://staging-api.resourcewatch.org/v1"
 geostore_api_url          = "https://staging-api.resourcewatch.org/v1"
 forms_api_url             = "https://staging-api.resourcewatch.org/v1"
 teams_api_url             = "https://dev-fw-api.globalforestwatch.org/v3"
+rw_areas_api_url          = "https://api.resourcewatch.org/v2"
```
Comment (owenjs) [inline terraform/vars/terraform-dev.tfvars]: Can you switch this to: `https://staging-api.resourcewatch.org/v2`

Can't mix and match between staging and production URL as the authentication will fail!

File: terraform/vars/terraform-staging.tfvars [hunk 4]
```diff
@@ -10,6 +10,7 @@ suppress_no_config_warning= "true"
 control_tower_url         = "https://staging-api.resourcewatch.org"
 areas_api_url             = "https://gfw-staging.globalforestwatch.org/v1"
 geostore_api_url          = "https://gfw-staging.globalforestwatch.org/v1"
+rw_areas_api_url          = "https://api.resourcewatch.org/v2"
```
Comment (owenjs) [inline terraform/vars/terraform-staging.tfvars]: Can you switch this to: `https://staging-api.resourcewatch.org/v2`

Can't mix and match between staging and production URL as the authentication will fail!

**Review by owenjs — CHANGES_REQUESTED** [review]

File: docker-compose-develop.yml [hunk 5]
```diff
@@ -18,6 +18,7 @@ services:
       LOCAL_URL: http://127.0.0.1:4400
       FORMS_API_URL: https://staging-api.resourcewatch.org/v1
       AREAS_API_URL: https://staging-api.resourcewatch.org/v1
+      RW_AREAS_API_URL: https://api.resourcewatch.org/v2
```
Comment (owenjs) [inline docker-compose-develop.yml]: and here for local testing `https://staging-api.resourcewatch.org/v2`

File: docker-compose-develop.yml [hunk 6]
```diff
@@ -44,6 +45,7 @@ services:
       AREAS_API_URL: http://127.0.0.1:9000
       GEOSTORE_API_URL: http://127.0.0.1:9000
       TEAMS_API_URL: "https://dev-fw-api.globalforestwatch.org/v3"
+      RW_AREAS_API_URL: https://api.resourcewatch.org/v2
```
Comment (owenjs) [inline docker-compose-develop.yml]: and here for local testing `https://staging-api.resourcewatch.org/v2`

**Review by owenjs — CHANGES_REQUESTED** [review]

**Review by owenjs — APPROVED** [review]

**Pull request merged**